### ISSUE TITLE
Or group

### ISSUE THREAD

**Issue by taras-strypko** [NRules/NRules#183]
Hi,

Not sure it's actually the bug - but when all conditions inside or group are met - action is executed for each condition.
For example:
```
.Or(x =>
                    x.Match<PurchaseConfiguration>(c => c.ProductCount == purchase.ProductCount)
                     .Match<PurchaseConfiguration>(c => purchase.Amount > c.Amount)
                );
```

In this case, if both conditions are met - my action is executed twice.
Is this expected behavior?
I was expecting, that in any case it shouldn't trigger action more than once.

If that's expected - is there any way to achieve logical 'or' between expressions ? (Except combining c# expressions). So, that if any of expression is true - action is executed (once).

Thanks

**Comment by snikolayev** [comment]
It’s expected, though I understand not very intuitive. Or groups are compiled to multiple rules behind the scenes, so unless the first firing of the rule invalidates the conditions, it will fire again if both groups matched.
There is an idea to implement exclusive or groups, but not high on the priority list.

**Comment by taras-strypko** [comment]
Ok, I see your point. Closing issue.
Thanks!
